File: app/report_store.py
"""Storage layer for generated reports."""

from __future__ import annotations

import json
from pathlib import Path
from typing import Any

STORAGE_PATH = Path("storage/generatedReports.json")


def ensure_store() -> None:
    if not STORAGE_PATH.exists():
        STORAGE_PATH.parent.mkdir(parents=True, exist_ok=True)
        STORAGE_PATH.write_text(json.dumps({"reports": [], "history": []}, indent=2), encoding="utf-8")


def read_store() -> dict[str, Any]:
    ensure_store()
    return json.loads(STORAGE_PATH.read_text(encoding="utf-8"))


def write_store(data: dict[str, Any]) -> None:
    ensure_store()
    STORAGE_PATH.write_text(json.dumps(data, indent=2), encoding="utf-8")


def append_report(report_record: dict[str, Any]) -> dict[str, Any]:
    store = read_store()
    store["reports"].append(report_record)
    store["history"].insert(
        0,
        {
            "reportId": report_record["id"],
            "generatedAt": report_record["generatedAt"],
            "templateId": report_record["templateId"],
            "title": report_record["title"],
            "exports": [entry["format"] for entry in report_record["exports"]],
        },
    )
    write_store(store)
    return report_record

File: src/reportStore.js
// DEPRECATED: Python FastAPI backend in app/ is the default runtime.
import fs from 'node:fs';
import path from 'node:path';

const storagePath = path.resolve('storage/generatedReports.json');

function ensureStore() {
  if (!fs.existsSync(storagePath)) {
    fs.writeFileSync(storagePath, JSON.stringify({ reports: [], history: [] }, null, 2));
  }
}

export function readStore() {
  ensureStore();
  return JSON.parse(fs.readFileSync(storagePath, 'utf-8'));
}

export function writeStore(data) {
  ensureStore();
  fs.writeFileSync(storagePath, JSON.stringify(data, null, 2));
}

export function appendReport(reportRecord) {
  const store = readStore();
  store.reports.push(reportRecord);
  store.history.unshift({
    reportId: reportRecord.id,
    generatedAt: reportRecord.generatedAt,
    templateId: reportRecord.templateId,
    title: reportRecord.title,
    exports: reportRecord.exports.map((entry) => entry.format)
  });
  writeStore(store);
  return reportRecord;
}
